module/zfs/dnode_sync.c

```diff
@@ -233,6 +235,7 @@ free_verify(dmu_buf_impl_t *db, uint64_t start, uint64_t end, dmu_tx_t *tx)
 		 * future txg.
 		 */
 		mutex_enter(&child->db_mtx);
+		rw_enter(&child->db_rwlock, RW_READER);
```

Comment:
As I understand, children are L0 here, so `db_rwlock` didn't promise to protect them.